utils.php

```diff
@@ -4029,3 +4029,44 @@ function sendSlackMessage($message) {
 
     curl_exec($curl);
 }
+
+function sendTestsolvingReminders() {
+    if (TESTSOLVE_REMINDER_HOURS == 0) {
+        return;
+    }
+
+    // Find puzzles that:
+    // - Are still in a draftable state (i.e. author can reasonably action them)
+    // - Have an editor assigned
+    // - Have not had a comment in the last week
+    // - Most recent comment was not from the author
+    $sql = sprintf("
+        SELECT
+            users.uid AS uid,
+            users.fullname AS user_fullname,
+            puzzles.id AS pid,
+            puzzles.title AS puzzle_title
+        FROM tester_links
+            LEFT JOIN users ON tester_links.uid = users.uid
+            LEFT JOIN puzzles ON tester_links.pid = puzzles.id
+        WHERE tester_links.created_at < DATE_SUB(CURRENT_TIMESTAMP(), INTERVAL '%d' HOUR)
+    ", TESTSOLVE_REMINDER_HOURS);
```

Comment:
There's no deduping here, right? So if this cronjob runs multiple times, it'll continue sending the same reminders. We could only run the cronjob daily but it would be nice if there was something more clever we could do.

(I also want to have Slack notifications when a puzzle is successfully testsolved, and all of my ideas for how to do that suffer from similar problems)